openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -22,22 +22,63 @@
                :class:`ParkerEtAl2020SSlab`
                :class:`ParkerEtAl2020SSlabB`
 """
+import os
 import math
-
 import numpy as np
+import pandas as pd
 from scipy.special import erf
 
 from openquake.baselib.general import CallableDict
 from openquake.hazardlib import const
 from openquake.hazardlib.gsim.base import GMPE, CoeffsTable, add_alias
 from openquake.hazardlib.imt import PGA, SA, PGV
 
+EPI_ADJS = os.path.join(os.path.dirname(__file__),
+                        "parker_2020_epi_adj_table.csv")
+
 CONSTANTS = {"b4": 0.1, "f3": 0.05, "Vb": 200,
              "vref_fnl": 760, "V1": 270, "vref": 760}
 
 _a0 = CallableDict()
 
 
+def _get_sigma_mu_adjustment(sat_region, trt, imt, epi_adjs_table):
+    """
+    Get the sigma_mu_adjustment (epistemic uncertainty) factor to be applied
+    to the mean ground-motion. Values are provided by authors in the
+    electronic for PGA and SA (not PGV) for each saturation regions.
+    """
+    # Map region to those within the adjustment table
+    if sat_region is None:
+        e_reg = 'Global'
+    elif sat_region in ['TW_N', 'TW_S']:
+        e_reg = 'Taiwan'
+    else:
+        e_reg = sat_region
+
+    # Get values from table for region and trt
+    adjs = epi_adjs_table.set_index('Region').loc[e_reg]
```

Comment:
It is best to set the index only once, when the CSV file is read and not millions of times as done here.